lib/IRGen/GenBuiltin.cpp

```diff
@@ -1105,7 +1105,42 @@ void irgen::emitBuiltinCall(IRGenFunction &IGF, const BuiltinInfo &Builtin,
     
     auto valueTy = getLoweredTypeAndTypeInfo(IGF.IGM,
                                              substitutions.getReplacementTypes()[0]);
-    
+
```

Comment:
For this we definitely need runtime testing.
I'm wondering if there are existing runtime tests for those builtins which we could run in embedded mode?

Reply:
Those tests should e.g. check for buffer overflow + underflow, etc.

Reply:
Good idea, agree, but I think I'll need to do that as a follow-up once we have some embedded stdlib building. Without it, it's very hard to write executable tests.

Reply:
Added a TODO